crates/evm/src/tests/ef_tests/cases/blockchain_test.rs

```diff
@@ -0,0 +1,291 @@
+                    gas_limit: case.genesis_block_header.gas_limit.to(),
+                    gas_used: case.genesis_block_header.gas_used.to(),
+                    timestamp: case.genesis_block_header.timestamp.to(),
+                    mix_hash: case.genesis_block_header.mix_hash,
+                    nonce: case.genesis_block_header.nonce.into(),
+                    base_fee_per_gas: case.genesis_block_header.base_fee_per_gas.map(|b| b.to()),
+                    extra_data: case.genesis_block_header.extra_data.clone(),
+                    // EIP-4844 related fields
+                    // https://github.com/Sovereign-Labs/sovereign-sdk/issues/912
+                    blob_gas_used: case.genesis_block_header.blob_gas_used.map(|b| b.to()),
+                    excess_blob_gas: case.genesis_block_header.excess_blob_gas.map(|b| b.to()),
+                    // EIP-4788 related field
+                    // unrelated for rollups
+                    parent_beacon_block_root: None,
+                };
+
+                let block = Block {
+                    header,
+                    l1_fee_rate: 0,
+                    l1_hash: B256::default(),
+                    transactions: 0u64..0u64,
+                };
+
+                for (&address, account) in case.pre.0.iter() {
+                    evm_config.data.push(AccountData::new(
+                        address,
+                        account.balance,
+                        account.code.clone(),
+                        account.nonce.saturating_to::<u64>(),
+                        HashMap::new(),
+                    ));
+                }
+
+                let (mut evm, _, mut storage) = get_evm_with_storage(&evm_config);
+
+                let mut working_set = WorkingSet::new(storage.clone());
+                evm.cfg.set(
+                    &EvmChainConfig {
+                        chain_id: evm_config.chain_id,
+                        limit_contract_code_size: evm_config.limit_contract_code_size,
+                        spec: vec![(0, SpecId::SHANGHAI)].into_iter().collect(),
+                        coinbase: case.genesis_block_header.coinbase,
+                        block_gas_limit: case.genesis_block_header.gas_limit.to(),
+                        base_fee_params: evm_config.base_fee_params,
+                    },
+                    &mut working_set,
+                );
+                evm.latest_block_hashes.set(
+                    &U256::from(0),
+                    &case.genesis_block_header.hash,
+                    &mut working_set,
+                );
+                evm.head.set(&block, &mut working_set);
+                evm.pending_head
+                    .set(&block, &mut working_set.accessory_state());
+                evm.finalize_hook(
+                    &case.genesis_block_header.state_root.0.into(),
+                    &mut working_set.accessory_state(),
+                );
```

Comment:
overriding what would normally be done in evm.genesis like this very smart. will make working with ef-tests easier